### ISSUE TITLE
How to pre-training base model by my own unsupervised dataset

### ISSUE THREAD

**Issue by ppaanngggg** [ultralytics/ultralytics#18290]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Current base models are all pre-trained from COCO dataset. I want to pre-train a base model specific for document pages. But I can't find how to pre-train from docs.

Please give me a hint, how to pre-train by ultralytics.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ppaanngggg, thank you for your interest in Ultralytics 🚀! We recommend visiting our [Docs](https://docs.ultralytics.com) where you can find detailed guides and examples on various topics, which might include solutions or pointers for your question about pre-training models.

If this is a custom training ❓ Question, we kindly ask you to provide additional details about your dataset and the specific approach you want to take. For instance, are you aiming for unsupervised pre-training using a specific technique? Sharing more details will help us and the community to assist you better. Also, make sure to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), which may provide valuable insights.

For real-time support, feel free to engage with our active community on [Discord](https://discord.com/invite/ultralytics) 🎧. If you have broader questions or ideas to discuss, you might find [Discourse](https://community.ultralytics.com) or our [Subreddit](https://reddit.com/r/Ultralytics) useful.

## Upgrade

To rule out any issues resolved in recent versions, ensure you're using the latest `ultralytics` release. Update your package and dependencies in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

To facilitate experimentation and testing, you can leverage verified environments for running YOLO:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Refer to [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. Refer to [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Follow our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) for setup <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing, validating all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu.

This is an automated response to ensure you get the help you need as soon as possible 😊! An Ultralytics engineer will review your question and provide further assistance soon. 🚀

**Comment by Burhan-Q** [comment]
"pre-train" would just be training on a dataset and then using the `best.pt` weights to train your "final" model. https://docs.ultralytics.com/modes/train

**Comment by ppaanngggg** [comment]
> "pre-train" would just be training on a dataset and then using the `best.pt` weights to train your "final" model. https://docs.ultralytics.com/modes/train

What I want to know is, is there any unsupervised way of training, just to learn features?

**Comment by glenn-jocher** [comment]
Ultralytics does not natively support unsupervised feature learning. However, you can explore self-supervised learning frameworks like SimCLR or BYOL to pre-train models and then fine-tune them with YOLO using supervised training. For YOLO-specific training, refer to https://docs.ultralytics.com/modes/train.

**Comment by ppaanngggg** [comment]
Thanks for your help

**Comment by glenn-jocher** [comment]
You're welcome! If you have any further questions, feel free to ask. The YOLO community and Ultralytics team are here to support you!

**Comment by Y-T-G** [comment]
You can try this:
https://y-t-g.github.io/tutorials/yolo-pretrain-ssl/
